lib/Cake/Model/Datasource/Database/Postgres.php

```diff
@@ -198,82 +198,138 @@ public function describe($model) {
 		$fields = parent::describe($table);
 		$this->_sequenceMap[$table] = array();
 		$cols = null;
-
 		if ($fields === null) {
-			$cols = $this->_execute(
-				"SELECT DISTINCT table_schema AS schema, column_name AS name, data_type AS type, is_nullable AS null,
-					column_default AS default, ordinal_position AS position, character_maximum_length AS char_length,
-					character_octet_length AS oct_length FROM information_schema.columns
-				WHERE table_name = ? AND table_schema = ?  ORDER BY position",
-				array($table, $this->config['schema'])
-			);
+            try {
+                $cols = $this->_execute(
+                    "WITH pg_attribute_local AS (
+                    SELECT * FROM pg_attribute
+                    WHERE attrelid = (? || '.' || ?)::regclass
+                    AND attnum > 0 )
+                SELECT
+                    nc.nspname::information_schema.sql_identifier AS schema,
+                    a.attname::information_schema.sql_identifier AS name,
+                    CASE
+                        WHEN t.typtype = 'd'::\"char\" THEN
+                        CASE
+                            WHEN bt.typelem <> 0::oid AND bt.typlen = '-1'::integer THEN 'ARRAY'::text
```

Comment:
Just pulled what information_schema was doing, my guess is it would work, but you want to make sure it's actually an int rather than getting implicitly cast to numeric or something, unlikely to be an issue, but the clarity is safer than letting it implicitly cast.